huaweicloud/resource_huaweicloud_compute_servergroup.go

```diff
@@ -77,19 +76,28 @@ func resourceComputeServerGroupCreate(d *schema.ResourceData, meta interface{})
 
 	membersToAdd := d.Get("members").(*schema.Set)
 	for _, v := range membersToAdd.List() {
+		instanceId := v.(string)
+		// The ECS instances do not support other operations when binding server groups.
+		config.MutexKV.Lock(instanceId)
+
 		var addMemberOpts servergroups.MemberOpts
-		addMemberOpts.InstanceID = v.(string)
+		addMemberOpts.InstanceID = instanceId
 		if err := servergroups.UpdateMember(ecsClient, addMemberOpts, "add_member", d.Id()).ExtractErr(); err != nil {
+			// Release the ECS instance after the binding fails.
+			config.MutexKV.Unlock(instanceId)
 			return fmtp.Errorf("Error to add an instance to ECS server group, err=%s", err)
 		}
+		// Release the ECS instance after the binding is complete.
+		config.MutexKV.Unlock(instanceId)
```

Comment:
```suggestion
		err := servergroups.UpdateMember(ecsClient, addMemberOpts, "add_member", d.Id()).ExtractErr();
		// Release the ECS instance after the binding option
		config.MutexKV.Unlock(instanceId)
			
		if err != nil {			
			return fmtp.Errorf("Error to add an instance to ECS server group, err=%s", err)
		}
```

how about writing like this?

Reply:
good idea